crypto/relic_build.sh

```diff
@@ -34,21 +39,61 @@ EP_METH=(-DEP_MIXED=ON -DEP_PLAIN=OFF -DEP_SUPER=OFF -DEP_DEPTH=4 -DEP_WIDTH=2 \
     -DEP_CTMAP=ON -DEP_METHD="JACOB;LWNAF;COMBS;INTER")
 PP_METH=(-DPP_METHD="LAZYR;OATEP")
 
-# Generate make files using cmake
-cmake "${COMP[@]}" "${GENERAL[@]}" \
+# "-march=native" is not supported on ARM by clang
+# we probe the cmake compiler detection logic by running cmake once with
+# the permissive logic, and adding the '-march=native' option only if
+# supported
+PROBE_COMP=(-DCOMP="-O3 -funroll-loops -fomit-frame-pointer -mtune=native")
+
+# make cmake print its CC interpretation
+echo 'message ( STATUS "CC=$ENV{CC}" )' >> ../CMakeLists.txt
+
+# Probe cmake's MakeFile generation
+CMAKE_OUTPUT=$(mktemp)
+cmake "${PROBE_COMP[@]}" "${GENERAL[@]}" \
     "${LIBS[@]}" "${RAND[@]}" \
     "${BN_REP[@]}" "${ARITH[@]}" \
     "${PRIME[@]}" "${PRIMES[@]}" \
     "${EP_METH[@]}" \
     "${BN_METH[@]}" \
     "${FP_METH[@]}" \
     "${FPX_METH[@]}" \
-    "${PP_METH[@]}" ..
+    "${PP_METH[@]}" .. > $CMAKE_OUTPUT
+# de-mangle the CMakeLists file, done with a temp file
+# to be portable between GNU / BSD sed
+CMAKE_TEMP=$(mktemp)
+sed -e '/message ( STATUS "CC=$ENV{CC}" )/d' ../CMakeLists.txt > $CMAKE_TEMP
+mv $CMAKE_TEMP ../CMakeLists.txt
+
+# extract what cmake uses as CC and probe its version string
+CC_VAL="$(tail -n 5 "$CMAKE_OUTPUT" | grep -oE 'CC=.*$' | sed -re 's/CC=(.*)/\1/g')"
+# default to which
+CC_VAL=${CC_VAL:-"$(which cc)"}
```

Comment:
Because that may not be the compiler used by cmake. In particular:
- this is configurable, e.g. you can pass CMAKE_C_COMPILER, CC, `-DCMAKE_C_COMPILER` ...
- `CMake` runs though its own [detection](https://gitlab.kitware.com/cmake/cmake/-/blob/v3.20.1/Modules/CMakeDetermineCCompiler.cmake) and then [testing](https://gitlab.kitware.com/cmake/cmake/-/blob/v3.20.1/Modules/CMakeTestCCompiler.cmake#L79) of the compiler
I didn't want to reproduce all that.

Reply:
Ah I understand, that makes sense